File: go/collections.go
package tachyon

import (
	"context"
	"net/http"
	"net/url"
)

// CollectionsService is /collections — create, list, and remove collections.
type CollectionsService struct {
	transport *httpTransport
}

// Create sends POST /collections. Field types are immutable after creation.
func (s *CollectionsService) Create(ctx context.Context, schema CollectionSchema) (CollectionInfo, error) {
	var out CollectionInfo
	err := s.transport.do(ctx, http.MethodPost, "/collections", requestOptions{body: schema}, &out)
	return out, err
}

// List sends GET /collections.
func (s *CollectionsService) List(ctx context.Context) ([]CollectionInfo, error) {
	var out []CollectionInfo
	err := s.transport.do(ctx, http.MethodGet, "/collections", requestOptions{}, &out)
	return out, err
}

// Retrieve sends GET /collections/{name}.
func (s *CollectionsService) Retrieve(ctx context.Context, name string) (CollectionInfo, error) {
	var out CollectionInfo
	err := s.transport.do(ctx, http.MethodGet, "/collections/"+url.PathEscape(name), requestOptions{}, &out)
	return out, err
}

// Delete sends DELETE /collections/{name}. Removes the collection and all its data.
func (s *CollectionsService) Delete(ctx context.Context, name string) error {
	return s.transport.do(ctx, http.MethodDelete, "/collections/"+url.PathEscape(name), requestOptions{}, nil)
}

File: typescript/src/collections.ts
import type { HttpClient } from './http.js';
import type { CollectionInfo, CollectionSchema } from './types.js';

/** `/collections` — create, list, and remove collections. */
export class CollectionsApi {
  constructor(private readonly http: HttpClient) {}

  /** `POST /collections`. Field types are immutable after creation. */
  create(schema: CollectionSchema): Promise<CollectionInfo> {
    return this.http.request<CollectionInfo>('POST', '/collections', { body: schema });
  }

  /** `GET /collections`. */
  list(): Promise<CollectionInfo[]> {
    return this.http.request<CollectionInfo[]>('GET', '/collections');
  }

  /** `GET /collections/{name}`. */
  retrieve(name: string): Promise<CollectionInfo> {
    return this.http.request<CollectionInfo>('GET', `/collections/${encodeURIComponent(name)}`);
  }

  /** `DELETE /collections/{name}`. Removes the collection and all its data. */
  async delete(name: string): Promise<void> {
    await this.http.request<void>('DELETE', `/collections/${encodeURIComponent(name)}`);
  }
}
